### DISCUSSION TITLE
New mod idea for osu!lazer and some other suggestions

### DISCUSSION THREAD

**Discussion by Biohazard34** [ppy/osu#21281] [Ideas]
This suggestion came from Tonio DND on Youtube. (not me)

1. New Mod: He suggested that a new mod be added that allowed for practicing specific sections of a map. Similar to Geometry Dash's practice mode, the mod will consist of settings of how it will behave during gameplay (ex. breaking SS, missing, or dying). But instead of restarting or failing you like SD and PF does, it will instead go to a bookmark in the map that the user can place or automatically be placed and resume from there. Hopefully toning down on the tediousness of practicing.

2. Collection Organization:  Another suggestion being adding organized collections similar to Windows' folder system and being able to put collections inside of collections and such. (Pretty sure this can already be done but I'm just reading off of his suggestion.)

3. Make filtering more user-friendly: And the final suggestion was adding a more user-friendly way of searching for specifc songs instead of typing  unplayed= star>6..bla bla blah.. and instead turn it into some type of context menu for filtering like having buttons and sliders which would be more easier to understand.

Again, this isn't my suggestion. I am simply posting on the behalf of the person that originally had these ideas. Still open for feedback on it however.

**Comment by mk56-spn** [comment]
Practice modes/ mods / options have been debated to death. Will come at some point. Although not  as a mod

**Comment by ghost** [comment]
I'm pretty sure beatmap search filters shown in a menu is duplicate 

e: found it #17494

**Comment by demon4dayz** [comment]
Referring to the mod idea, I think a "TAS" or "Segmented Speedrun" type mod would be welcomed with open arms by the community. Here's how it would work in my opinion: 

It would work like TAG Coop, where each combo color change would be a "save state" of sort, where the player would go back to the beginning of the combo color upon missing. This would also solve the issue of the reset being too abrupt (so the player would see the previous combo color being hit by a 'ghost' replay of themselves from earlier. Maybe an option to skip the replay could also be added if the previous combo color is too long like a deathstream).

The play would finish when either the map ends or the player quits playing the map. In terms of score/pp I think it should be treated like RX or AP as both mods assist the player in achieving their score.

The mod could also have the same properties as SD and PF where the player could toggle if the reset happens on combo break or accuracy drop.
